vpr/src/route/rr_graph2.cpp

```diff
@@ -1121,6 +1141,116 @@ static void load_chan_rr_indices(const int max_chan_width,
     }
 }
 
+vtr::NdMatrix<int,2> get_number_track_to_track_inter_die_conn(vtr::NdMatrix<t_inter_die_switchblock_edge, 5>& multi_layer_track_conn,
+                                              t_sb_connection_map* sb_conn_map,
+                                              RRGraphBuilder& rr_graph_builder) {
+
+    auto& grid_ctx = g_vpr_ctx.device().grid;
+    vtr::NdMatrix<int, 2> extra_nodes_per_switchblocks;
+    extra_nodes_per_switchblocks.resize(std::array<size_t, 2>{grid_ctx.width(), grid_ctx.height()});
+
+    for (size_t y = 0; y < grid_ctx.height(); y++) {
+        for (size_t x = 0; x < grid_ctx.width(); x++) {
+            int curr_switchblocks_offset =0;
+            for (auto layer = 0; layer < grid_ctx.get_num_layers(); layer++){
+                for (auto from_side : TOTAL_SIDES) {
+                    for (auto to_side : TOTAL_SIDES) {
+                        if (from_side < NUM_SIDES && to_side < NUM_SIDES) { //this connection is not crossing any layer
+                            continue;
+                        } else if (from_side == to_side) {
+                            continue;
+                        } else {
+                            Switchblock_Lookup sb_coord(x, y, layer, from_side, to_side);
+                            if (sb_conn_map->count(sb_coord) > 0) {
+                                std::vector<t_switchblock_edge>& conn_vector = (*sb_conn_map)[sb_coord];
+                                for (int iconn = 0; iconn < (int)conn_vector.size(); ++iconn) {
+                                    if (conn_vector[iconn].to_wire_layer == conn_vector[iconn].from_wire_layer) {
+                                        //this connection doesn't cross any layer, no new node is required
+                                        continue;
+                                    } else {
+                                        //check if both from_node and to_node exists in the rr-graph
+
+                                        //CHANY -> CHANX connection
+                                        RRNodeId from_inode = rr_graph_builder.node_lookup().find_node(conn_vector[iconn].from_wire_layer, x, y, CHANY, conn_vector[iconn].from_wire_layer);
```

Comment:
I'd have a variable chany_index = 1 and chanx_index = 0 to make this easier to read.
May be some code duplication that a helper function could remove.